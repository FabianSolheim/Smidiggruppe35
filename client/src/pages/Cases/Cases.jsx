import React from "react";
import { Box, Container, Grid, Typography } from "@mui/material";
import { sectors } from "../../components/SectorCard/sectors";
import { SectorCard } from "../../components/SectorCard/sectorCard";
import { HeadlineWithUnderline } from "../../components/HeadlineWithUnderline/headlineWithUnderline";

export function Cases() {
  return (
    <>
      <div style={{ marginTop: 50 }} />
      {HeadlineWithUnderline("Cases", 80, "#034F7A", 0, "#212121", "36px")}
      <Grid container justifyContent="center" paddingTop={5}>
        <Grid item xs={12} sm={4}>
          <Container>
            A lot of giving platforms leave us with the paradox of choice. Many
            corporate giving systems don't thoroughly vet the projects and
            nonprofits that are featured.
            <br />
            <br />
            Our platform has two categories: knowledge and water. We know that
            these two areas are vital to thriving societiesâ€”this is why our
            portfolio focuses on them.
          </Container>
        </Grid>
      </Grid>
      <Box paddingBottom={{ xs: 5, md: 15 }} px={{ xs: 2 }} paddingTop={5}>
        <Grid container justifyContent="center" spacing={6}>
          {sectors.map((sector) => (
            <Grid item lg={4.1} xs={12} sm={6} md={5}>
              <SectorCard
                key={sector.title}
                title={sector.title}
                content={sector.content}
                image={sector.image}
                linkTo={sector.linkTo}
              />
            </Grid>
          ))}
        </Grid>
      </Box>
    </>
  );
}
