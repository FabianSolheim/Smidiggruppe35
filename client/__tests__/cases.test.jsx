import ReactDOM from "react-dom";
import React from "react";
import { Cases } from "../src/pages/Cases/Cases.jsx";
import { MemoryRouter } from "react-router-dom";

describe("btnWithHeadline test", () => {
  it("shows btnWithHeadline component", () => {
    const domElement = document.createElement("div");
    ReactDOM.render(
      <MemoryRouter>
        <Cases />
      </MemoryRouter>,
      domElement
    );
    expect(domElement.innerHTML).toMatchSnapshot();
  });
});
