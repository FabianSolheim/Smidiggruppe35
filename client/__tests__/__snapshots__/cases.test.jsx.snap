// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`btnWithHeadline test shows btnWithHeadline component 1`] = `
"<div style=\\"margin-top: 80px;\\"></div><div><div style=\\"font-size: 36px; text-align: center; color: rgb(33, 33, 33); font-family: Josefin Sans; font-weight: 500;\\">Cases</div></div><div style=\\"width: 80px; height: 2px; background-color: rgb(73, 3, 116); margin: 5px auto 0px auto;\\"></div><div class=\\"MuiGrid-root MuiGrid-container css-1jtov7i-MuiGrid-root\\"><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 css-1b3l6lk-MuiGrid-root\\"><div class=\\"MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root\\">A lot of giving platforms leave us with the paradox of choice. Many corporate giving systems don't thoroughly vet the projects and nonprofits that are featured.<br><br>Our platform has two categories: knowledge and water. We know that these two areas are vital to thriving societiesâ€”this is why our portfolio focuses on them.</div></div></div><div class=\\"MuiBox-root css-1tjwps2\\"><div class=\\"MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-6 css-bmgubx-MuiGrid-root\\"><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-5 MuiGrid-grid-lg-4.1 css-q9ajfo-MuiGrid-root\\"><a href=\\"/water\\"><div class=\\"sector-container MuiBox-root css-i3nlsl\\"><div class=\\"MuiGrid-root MuiGrid-container css-1hqcngf-MuiGrid-root\\"><div class=\\"MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root\\"><img src=\\"https://i.imgur.com/pi1w9X5.png\\" class=\\"sector-image\\"></div><div class=\\"MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root\\"><div class=\\"MuiBox-root css-1yuhvjn\\"><h3 class=\\"MuiTypography-root MuiTypography-h3 MuiTypography-gutterBottom sector-header css-1kn7tn9-MuiTypography-root\\">Water</h3><p class=\\"MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter sector-content css-95p7p-MuiTypography-root\\">Imagine spending hours everyday searching for water. 
but with clean water, everything changes</p><div class=\\"sector-button MuiBox-root css-zw9846\\"><a class=\\" css-vcbgxq-MuiButton-root\\" component=\\"button\\" focusvisibleclassname=\\"\\" style=\\"padding: 12px 22px; text-decoration: none; font-weight: 500; border-radius: 0; text-transform: none;\\" classes=\\"[object Object]\\" href=\\"/water\\">Case study</a></div></div></div></div></div></a></div><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-5 MuiGrid-grid-lg-4.1 css-q9ajfo-MuiGrid-root\\"><a href=\\"/knowledge\\"><div class=\\"sector-container MuiBox-root css-i3nlsl\\"><div class=\\"MuiGrid-root MuiGrid-container css-1hqcngf-MuiGrid-root\\"><div class=\\"MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root\\"><img src=\\"https://i.imgur.com/ltNtgN8.png\\" class=\\"sector-image\\"></div><div class=\\"MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root\\"><div class=\\"MuiBox-root css-1yuhvjn\\"><h3 class=\\"MuiTypography-root MuiTypography-h3 MuiTypography-gutterBottom sector-header css-1kn7tn9-MuiTypography-root\\">Knowledge</h3><p class=\\"MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter sector-content css-95p7p-MuiTypography-root\\">Knowledge is one of the greatest tools for 
change, one of the most important factors to 
fight the climate crisis, and one of the best ways 
to break the cycle of poverty</p><div class=\\"sector-button MuiBox-root css-zw9846\\"><a class=\\" css-vcbgxq-MuiButton-root\\" component=\\"button\\" focusvisibleclassname=\\"\\" style=\\"padding: 12px 22px; text-decoration: none; font-weight: 500; border-radius: 0; text-transform: none;\\" classes=\\"[object Object]\\" href=\\"/knowledge\\">Case study</a></div></div></div></div></div></a></div></div></div>"
`;
